wall_e/src/WalleModels/customFields.py

```diff
@@ -0,0 +1,36 @@
+from django.db import models
```

Comment:
why is this file needed?

Reply:
django didn't support fixed char fields or the identity column. the identity column support is [on the dev channel](https://code.djangoproject.com/ticket/30511), but im not waiting for that . why identity column instead of serial? idk, i came across this [stackoverflow](https://stackoverflow.com/a/55300741) post while googling something i cant remember and since then i've used generated as identity as the column type for ids. 

i was using the fixed char to store the discord id's since they're exactly 11 chars long but i was converting back and forth between and int and string so i dropped using it, but kept the code for the custom type incase it'll be useful.

Reply:
regarding the `GeneratedIdentityField` SO post, that's sure an interesting read that I don't fully understand but ok. i am not against using newer conventions, when when I don't know what they do 😛 . but are you able to link me to anything that indicates that `GeneratedIdentityField` that mimic what Identity keys are supposed to do?

but please delete the `FixedCharField`, I have no interest in keeping around a field **in case** we need it.

But, is there a reason why this is a separate file instead of in `wall_e/src/WalleModels/models.py`?

Reply:
removed fixedcharfield. I put the custom types in their own files because it made sense to keep the database models separate from custom field definitions. 

I dont know what you mean by `identity key`.

Reply:
>  I put the custom types in their own files because it made sense to keep the database models separate from custom field definitions.

Fair enough

> I dont know what you mean by identity key.

I mean the `Identity column`, I assume each entry is referred to as an `identity key`

Reply:
few links that may have what you're looking for: 
https://en.wikipedia.org/wiki/Identity_column
https://www.postgresqltutorial.com/postgresql-identity-column/
https://www.geeksforgeeks.org/postgresql-identity-column/

the underlying type is just `integer`. i've never heard it referred to as an `identity key`, just id column or pk(if used as a pk), i guess if you had an identity column that was the pk you would just refer to it by its column name since it would probably have some other purpose.

Reply:
this is what I was looking for https://docs.djangoproject.com/en/4.0/howto/custom-model-fields/#custom-database-types